src/main/java/io/github/jhipster/registry/config/JWTSecurityConfiguration.java

```diff
@@ -4,32 +4,85 @@
 import io.github.jhipster.registry.security.Http401UnauthorizedEntryPoint;
 import io.github.jhipster.registry.security.jwt.JWTConfigurer;
 import io.github.jhipster.registry.security.jwt.TokenProvider;
+import org.springframework.beans.factory.BeanInitializationException;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
 import org.springframework.context.annotation.Profile;
 import org.springframework.http.HttpMethod;
-import org.springframework.security.config.annotation.method.configuration.EnableGlobalMethodSecurity;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.config.annotation.authentication.builders.AuthenticationManagerBuilder;
 import org.springframework.security.config.annotation.web.builders.HttpSecurity;
 import org.springframework.security.config.annotation.web.builders.WebSecurity;
 import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
 import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
 import org.springframework.security.config.http.SessionCreationPolicy;
+import org.springframework.security.core.userdetails.User;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.security.provisioning.InMemoryUserDetailsManager;
 
 @Configuration
 @EnableWebSecurity
-@EnableGlobalMethodSecurity(prePostEnabled = true, securedEnabled = true)
 @Profile("!" + Constants.PROFILE_OAUTH2)
 public class JWTSecurityConfiguration extends WebSecurityConfigurerAdapter {
 
+    private final AuthenticationManagerBuilder authenticationManagerBuilder;
+
     private final Http401UnauthorizedEntryPoint authenticationEntryPoint;
 
     private final TokenProvider tokenProvider;
 
-    public JWTSecurityConfiguration(Http401UnauthorizedEntryPoint authenticationEntryPoint,
-                                 TokenProvider tokenProvider) {
+    private final String username;
+
+    private final String password;
+
+    private final String[] roles;
+
+    public JWTSecurityConfiguration(@Value("${spring.security.user.name}") String username,
```

Comment:
This is the most important change. I'm not sure about what I do here, as I didn't found a way to use directly `spring.security.user.name` with Basic Auth. That's why I need to use the properties here and create the bean `userDetailsService`